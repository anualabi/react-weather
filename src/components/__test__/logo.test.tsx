import { render, screen } from "@testing-library/react";

import Logo from "@/components/logo";

describe("Logo", () => {
  it("should render (snapshot)", () => {
    const { asFragment } = render(<Logo />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render a level 2 heading", () => {
    render(<Logo />);
    const heading = screen.getByRole("heading", { level: 2 });
    expect(heading).toBeInTheDocument();
  });

  it("should display the correct text content in the heading", () => {
    render(<Logo />);
    const heading = screen.getByRole("heading", { level: 2 });
    expect(heading).toHaveTextContent("React Weather");
  });

  it("should have a link pointing to the homepage", () => {
    render(<Logo />);
    const link = screen.getByRole("link");
    expect(link).toHaveAttribute("href", "/");
  });
});
