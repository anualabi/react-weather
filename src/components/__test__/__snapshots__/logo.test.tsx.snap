// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo should render (snapshot) 1`] = `
<DocumentFragment>
  <a
    href="/"
  >
    <h2
      class="scroll-m-20 text-2xl md:text-3xl xl:text-4xl text-white font-black capitalize"
    >
      React Weather
    </h2>
  </a>
</DocumentFragment>
`;
